common/txmgr/types/tx.go

```diff
@@ -31,17 +31,43 @@ type TxStrategy interface {
 	PruneQueue(pruneService UnstartedTxQueuePruner, qopt pg.QOpt) (n int64, err error)
 }
 
-type TxAttemptState string
+type TxAttemptState int8
 
 type TxState string
 
 const (
-	TxAttemptInProgress = TxAttemptState("in_progress")
-	// TODO: Make name chain-agnostic (https://smartcontract-it.atlassian.net/browse/BCI-981)
-	TxAttemptInsufficientEth = TxAttemptState("insufficient_eth")
-	TxAttemptBroadcast       = TxAttemptState("broadcast")
+	TxAttemptInProgress TxAttemptState = iota + 1
+	TxAttemptInsufficientFunds
+	TxAttemptBroadcast
 )
 
+func NewTxAttemptState(state string) (s TxAttemptState) {
+	switch state {
+	case "in_progress":
```

Comment:
These string values are duplicated in these 2 functions.
Could we use a var for string names, and use that in both these functions?

Or, is there any reason you prefer int8 instead of string?